src/x3f_io.c

```diff
@@ -78,6 +78,12 @@ static int x3f_get4(FILE *f)
 #define GET1(_v) do {(_v) = x3f_get1(I->input.file);} while (0)
 #define GET2(_v) do {(_v) = x3f_get2(I->input.file);} while (0)
 #define GET4(_v) do {(_v) = x3f_get4(I->input.file);} while (0)
+#define GET4F(_v)				\
+  do {						\
+    union {int32_t i; float f;} _tmp;		\
+    _tmp.i = x3f_get4(I->input.file);		\
+    (_v) = _tmp.f;				\
+  } while (0)
```

Comment:
It has to do with a syntax mistake in C.

Lets say you want to be able to write
if (test) A();

Then this will result in compilation error
#define A() {a,b,c}
but this works
#define A() do {a,b,c} while(0)

Of course, you could do
#define A() a,b,c
but the a,b,c is not in its own block, which is problematic
if you e.g. want to declare local variables.